kasa/smart/modules/powerprotection.py

```diff
@@ -0,0 +1,105 @@
+"""Power protection module."""
+
+from __future__ import annotations
+
+from ...feature import Feature
+from ..smartmodule import SmartModule
+
+
+class PowerProtection(SmartModule):
+    """Implementation for power_protection."""
+
+    REQUIRED_COMPONENT = "power_protection"
+
+    def _initialize_features(self) -> None:
+        """Initialize features after the initial update."""
+        self._add_feature(
+            Feature(
+                device=self._device,
+                id="overloaded",
+                name="Overloaded",
+                container=self,
+                attribute_getter="overloaded",
+                type=Feature.Type.BinarySensor,
+                category=Feature.Category.Info,
+            )
+        )
+        self._add_feature(
+            Feature(
+                device=self._device,
+                id="power_protection_enabled",
+                name="Power protection enabled",
+                container=self,
+                attribute_getter="enabled",
+                attribute_setter="set_enabled",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+        self._add_feature(
+            Feature(
+                device=self._device,
+                id="power_protection_threshold",
+                name="Power protection threshold",
+                container=self,
+                attribute_getter="protection_threshold",
+                attribute_setter="set_protection_threshold",
+                unit_getter=lambda: "W",
+                type=Feature.Type.Number,
+                range_getter="protection_threshold_range",
+                category=Feature.Category.Config,
+            )
+        )
+
+    def query(self) -> dict:
+        """Query to execute during the update cycle."""
+        return {"get_protection_power": None, "get_max_power": None}
+
+    @property
+    def overloaded(self) -> bool:
+        """Return True is power protection has been triggered.
+
+        This value remains True until the device is turned on again.
+        """
+        return self._device.sys_info["power_protection_status"] == "overloaded"
+
+    @property
+    def enabled(self) -> bool:
+        """Return True if child protection is enabled."""
+        return self.data["get_protection_power"]["enabled"]
+
+    async def set_enabled(self, enabled: bool) -> dict:
+        """Set child protection."""
+        params = {**self.data["get_protection_power"], "enabled": enabled}
```

Comment:
From looking at the fixtures all the devices with `enabled` False seem to have a default protection power of 0. This means that in order to enable power protection consumers would have to make two calls which doesn't seem right.  Perhaps it would be better to have the module expose a single setter that allows setting `threshold` and `enabled` at the same time. The feature could then just be a `Number` and the setter will automatically (via a lambda or small def) set `enabled = threshold > 0`.

Reply:
Hmm, sounds fair enough & avoids creating a separate switch for homeassistant which is good 👍